.gitignore

```diff
@@ -0,0 +1,112 @@
+# common C ignores
```

Comment:
Do we need/want a copyright header for this file? 
I know we didn't have one before, but should we have one?

Reply:
In my opinion, .gitignore is a configuration file and therefore I would vote against a copyright header.

Reply:
fine with me, any other opinions @juergenchrist @hfreude